### PR TITLE
Renesas: Smartbond: Update MIPI DBI driver to work with PM (update)

### PR DESCRIPTION
**Pull request by ioannis-karachalios**
This PR should be a continuation of #72184

### CODE REVIEW

**Review by danieldegrasse — COMMENTED** [review]
Can you make the commit title a bit more descriptive? Perhaps `drivers: display: smartbond: acquire power rail when using PM`?

**Comment by ioannis-karachalios** [comment]
> Can you make the commit title a bit more descriptive? Perhaps `drivers: display: smartbond: acquire power rail when using PM`?

Done

**Review by blauret — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

File: drivers/mipi_dbi/mipi_dbi_smartbond.c [hunk 1]
```diff
@@ -497,11 +497,24 @@ static int mipi_dbi_smartbond_pm_action(const struct device *dev, enum pm_device
...
 			ret = 0;
 		}
 		break;
 	case PM_DEVICE_ACTION_RESUME:
 		__ASSERT_NO_MSG(!data->is_active);
+
+		/*
+		 * Although PD_SYS should already be turned on, make sure LCD controller's
+		 * power domain is up and running before accessing the display block.
+		 * Acquiring PD_COM is mandatory when in PM runtime mode.
```
Comment (ydamigos) [inline drivers/mipi_dbi/mipi_dbi_smartbond.c:514]:
```suggestion
		 * Acquiring PD_SYS is mandatory when in PM runtime mode.
```

File: drivers/mipi_dbi/mipi_dbi_smartbond.c [hunk 2]
```diff
@@ -497,11 +497,24 @@ static int mipi_dbi_smartbond_pm_action(const struct device *dev, enum pm_device
 		/* Sleep is only allowed when there are no active LCDC operations */
 		if (!data->is_active) {
 			(void)mipi_dbi_smartbond_suspend(dev);
+			/*
+			 * Once the display block is turned off, its power rail
```
Comment (ydamigos) [inline drivers/mipi_dbi/mipi_dbi_smartbond.c:501]:
```suggestion
			 * Once the display block is turned off, its power domain
```

**Review by ydamigos — APPROVED** [review]

**Review by andrzej-kaczmarek — APPROVED** [review]

**Review by blauret — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**